src/lib/MongoQueue.ts

```diff
@@ -0,0 +1,76 @@
+import { Model, Schema } from 'mongoose';
+import { Queue, Media, QueueManager } from 'musicca';
+
+export interface IQueue {
+  _id: string;
+  list: Media[];
+}
+
+export const schema = new Schema<IQueue>({
+  _id: String,
+  list: {
+    type: [Object],
+    default: []
+  }
+});
+
+export default class MongoQueue extends Queue {
+  public db: Model<IQueue>;
+
+  public id: string;
+
+  constructor(model: Model<IQueue>, manager: QueueManager<Queue>, id: string) {
+    super(manager, id);
+    this.id = id;
+    // eslint-disable-next-line new-cap
+    new model({
+      _id: this.id,
+      list: []
+    }).save();
+    this.db = model;
+  }
+
+  public async all() {
+    const a = await this.db.findById(this.id);
+    return a?.list ?? [];
```

Comment:
Please convert the elements back to the `Media` object. You could store the extractor id then fetch them back here.